### PR TITLE
Initial commit for supporting SSM OpsMetadata

### PR DESCRIPTION
**Pull request by sahithj**
*Issue #, if available:*

*Description of changes:*
Initial commit for supporting SSM OpsMetadata

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-ssm-opsmetadata/aws-ssm-opsmetadata.json [hunk 1]
```diff
@@ -0,0 +1,127 @@
+{
+    "typeName": "AWS::SSM::OpsMetadata",
+    "description": "Resource Type definition for AWS::SSM::OpsMetadata",
+    "sourceUrl": "https://github.com/aws-cloudformation/aws-cloudformation-rpdk.git",
+    "definitions": {
+        "MetadataValueString": {
+            "$comment": "Metadata Value",
+            "type": "string",
```
Comment (nbharad) [inline aws-ssm-opsmetadata/aws-ssm-opsmetadata.json:8]: Can you please let me know if pattern validation can be added?

**Review by nbharad — APPROVED** [review]

File: aws-ssm-opsmetadata/aws-ssm-opsmetadata.json [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        "/properties/OpsMetadataArn"
+    ],
+    "createOnlyProperties": [
+        "/properties/ResourceId"
+    ],
+    "handlers": {
+        "create": {
+            "permissions": [
+                "ssm:CreateOpsMetadata",
+                "ssm:GetOpsMetadata"
```
Comment (adiga-aws) [inline aws-ssm-opsmetadata/aws-ssm-opsmetadata.json:97]: do we need to include tag related permissions here?

File: aws-ssm-opsmetadata/aws-ssm-opsmetadata.json [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                "ssm:AddTagsToResource",
+                "ssm:RemoveTagsFromResource",
+                "ssm:ListTagsForResource",
+                "ssm:GetOpsMetadata"
+            ],
+            "timeoutInMinutes": 5
+        },
+        "delete": {
+            "permissions": [
+                "ssm:DeleteOpsMetadata"
```
Comment (adiga-aws) [inline aws-ssm-opsmetadata/aws-ssm-opsmetadata.json:118]: should this include `ssm:RemoveTagsFromResource` as well?

**Review by adiga-aws — APPROVED** [review]

File: aws-ssm-opsmetadata/aws-ssm-opsmetadata.json (hunk 2 above)
Comment (sahithj) [reply aws-ssm-opsmetadata/aws-ssm-opsmetadata.json:97]: Tag related APIs have been added to update only and this is required when a stack is updated to add/remove tags.

File: aws-ssm-opsmetadata/aws-ssm-opsmetadata.json (hunk 3 above)
Comment (sahithj) [reply aws-ssm-opsmetadata/aws-ssm-opsmetadata.json:118]: It is not required as the service takes care of deleting the tags.

File: aws-ssm-opsmetadata/aws-ssm-opsmetadata.json (hunk 1 above)
Comment (sahithj) [reply aws-ssm-opsmetadata/aws-ssm-opsmetadata.json:8]: Validation is taken care by the service.

**Pull request merged**